Automattic/pocket-casts-android
# Add chapters to episode details screen

## Description

This PR adds chapters preview to episode details. It also improves couple of other things:
- Data set changes in player container are now applied selectively with appropriate `notifyItemInserted()` and `notifyItemRemoved()`.
- Chapters are now available in player view before playing the episode. They are not extracted beforehand so embedded chapters still require playing but once extracted and saved in the database they will be there before playing.
- I removed snackbar feedback about deselecting last chapter because it is now blocked in the UI.
- `ChaptersViewModel` is now extensively tested.

## Testing Instructions

Test both episode details and player chapters. They should behave almost the same. The only difference is that tapping on an already playing chapter in the episode details doesn't navigate to player. Check if upsell still works.

Test player behavior with tabs. Add to the queue episodes with and without chapters and skip between them. Chapters tab should behave correctly and appear or disappear depending on whether episode has chapters.

## Screenshots or Screencast 

| Light | Dark | Rose | Indigo | Extra dark |
| - | - | - | - | - |
| ![Screenshot_20240606-120624](https://github.com/Automattic/pocket-casts-android/assets/30936061/e7226301-e940-4084-b8bf-f76193ebabea) | ![Screenshot_20240606-120650](https://github.com/Automattic/pocket-casts-android/assets/30936061/92dcf2d9-e138-49a3-b193-c5096c41f849) | ![Screenshot_20240606-120700](https://github.com/Automattic/pocket-casts-android/assets/30936061/0bf050b9-e6a9-43fc-96b0-8a917052e0e6) | ![Screenshot_20240606-120713](https://github.com/Automattic/pocket-casts-android/assets/30936061/38c17ddd-aa89-429f-8ae0-e47509ab09f2) | ![Screenshot_20240606-120727](https://github.com/Automattic/pocket-casts-android/assets/30936061/234b6431-9a79-4e31-90fd-c71f611973d1) |

| Dark Contrast | Light Contrast | Electricity | Classic | Radioactivity |
| - | - | - | - | - |
| ![Screenshot_20240606-120739](https://github.com/Automattic/pocket-casts-android/assets/30936061/59bb556f-bec4-49f3-a576-06984f44377d) | ![Screenshot_20240606-120750](https://github.com/Automattic/pocket-casts-android/assets/30936061/d3e84eee-0b16-48a4-84ce-c0c262d584f9) | ![Screenshot_20240606-120759](https://github.com/Automattic/pocket-casts-android/assets/30936061/f0bbcf80-e4b0-419c-b1cc-5b2de5fc42a9) | ![Screenshot_20240606-120812](https://github.com/Automattic/pocket-casts-android/assets/30936061/814d113c-9f20-47a8-a332-5f562dbec6d8) | ![Screenshot_20240606-120832](https://github.com/Automattic/pocket-casts-android/assets/30936061/5e0d7a8b-1fbc-47ed-9559-a3148e2d8ee3) |

## Checklist
 
#### I have tested any UI changes...
